src/util/Utils.cpp

```diff
@@ -276,14 +247,48 @@ int Utils::getFileSize(std::string filePath) {
     return fileSize;
 }
 
+/**
+ * This method compresses files using gzip
+ * @param filePath
+ */
+void Utils::compressFile(const std::string filePath, const std::string mode) {
+    char buffer[128];
+    std::string result = "";
+    std::string command = mode + " -f " + filePath + " 2>&1";
+    char *commandChar = new char[command.length() + 1];
+    strcpy(commandChar, command.c_str());
+    FILE *input = popen(commandChar, "r");
+    if (input) {
+        while (!feof(input)) {
+            if (fgets(buffer, 128, input) != NULL) {
+                result.append(buffer);
+            }
+        }
+        pclose(input);
+        if (!result.empty()) {
+            if (result.find("pigz: not found") != std::string::npos) {
+                util_logger.log("pigz not found. Compressing using gzip", "info");
+                compressFile(filePath, "gzip");
+            } else {
+                util_logger.log("File compression failed with error: " + result, "error");
+            }
+        } else {
+            //util_logger.log("File in " + filePath + " compressed with gzip", "info");
+        }
+    } else {
+        perror("popen");
```

Comment:
Throwing an error like this way does not make much sense. It should be with a meaningful message.